### PR TITLE
Kj/service definition

### PR DESCRIPTION
**Pull request by Katinka666**

### CODE REVIEW

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java [hunk 1]
```diff
@@ -55,6 +58,37 @@ public List<ServiceDefinition> createBulk(final List<String> names) {
...
+		} catch (final Exception ex) {
+			logger.error(ex.getMessage());
+			logger.debug(ex);
+			throw new InternalServerError("Database operation error");
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public Page<ServiceDefinition> getPage(final PageRequest pagination) {
+		logger.debug("getServiceDefinitionEntities started...");
```
Comment (rbocsi) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java]: not the method name is referenced here

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java [hunk 2]
```diff
@@ -55,6 +58,37 @@ public List<ServiceDefinition> createBulk(final List<String> names) {
...
+
+		synchronized (LOCK) {
```
Comment (rbocsi) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java]: I don't see a reason to use a LOCK just for reading from the db.

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/dto/DTOConverter.java [hunk 3]
```diff
@@ -70,14 +70,23 @@ public DeviceResponseDTO convertDeviceEntityToDeviceResponseDTO(final Device dev
...
 
 		final List<ServiceDefinitionResponseDTO> converted = entities.stream()
 				.map(e -> convertServiceDefinitionEntityToDTO(e))
 				.collect(Collectors.toList());
 		return new ServiceDefinitionListResponseDTO(converted, converted.size());
 	}
 
+	//-------------------------------------------------------------------------------------------------
+	public ServiceDefinitionListResponseDTO convertServiceDefinitionEntityPageToDTO(final Page<ServiceDefinition> entities) {
+		logger.debug("convertServiceDefinitionEntityListToDTO started...");
```
Comment (rbocsi) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/dto/DTOConverter.java]: not matching with the method name

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ManagementValidation.java [hunk 4]
```diff
@@ -216,30 +226,86 @@ public List<String> validateAndNormalizeRemoveDevices(final List<String> names,
 	// SERVICE DEFINITION VALIDATION
 
 	//-------------------------------------------------------------------------------------------------
-	public void validateCreateServiceDefinition(final ServiceDefinitionListRequestDTO dto, final String origin) {
+	public void validateQueryServiceDefinitions(final PageDTO dto, final String origin) {
+		logger.debug("validateGetServiceDefinitions started");
```
Comment (rbocsi) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ManagementValidation.java]: not matching with the method name

File: serviceregistry/src/test/java/eu/arrowhead/serviceregistry/service/validation/NameValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	// members
+
+	@Autowired
+	private NameValidator nameValidator;
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Disabled
```
Comment (rbocsi) [inline serviceregistry/src/test/java/eu/arrowhead/serviceregistry/service/validation/NameValidatorTest.java]: no need to disable the existing tests

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java [hunk 6]
```diff
@@ -67,16 +67,13 @@ public List<ServiceDefinition> createBulk(final List<String> names) {
...
-
-		synchronized (LOCK) {
-			try {
-				return repo.findAll(pagination);
-			} catch (final Exception ex) {
-				logger.error(ex.getMessage());
-				logger.debug(ex);
-				throw new InternalServerError("Database operation error");
-			}
+		logger.debug("getPage started...");
```
Comment (rbocsi) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/ServiceDefinitionDbService.java:68]: It is good that you removed the syncronized block, but now you have an unused LOCK member, which should be removed as well.

**Review by rbocsi — APPROVED** [review]
Approved, but I found one small issue which should be fixed.

**Review by borditamas — APPROVED** [review]

**Pull request merged**